Example: Ping an App via an internal address (#741)
superfly/docs: reference/private-networking.html.md

Examples of retrieving this information are in the [fly-examples/privatenet](https://github.com/fly-apps/privatenet) repository.

> ### Example: Ping an app via an `.internal` address

Reviewer: "Reach" an app? Since you use curl, not ping

A quick to tutorial to show private networks in action. If you've arrived on private networking, we'll assume you've at least installed the CLI and are at least a little familiar with deploying apps on Fly.

For this example we'll use the [hello-fly](https://github.com/fly-apps/hello-fly) repository.

```cmd
# clone the repository
git clone https://github.com/fly-apps/hello-fly

cd hello-fly

# create an app on fly
fly launch

# output omitted...

? Would you like to deploy now? (y/N)
N

# deploying now is fine, but we'll want to modify
# the fly.toml and redeploy again
```

> #### Expose an `internal_port` in the fly.toml

Reviewer 2: When I run `fly launch` on `hello-fly` my `fly.toml` looks like this:
```
app = "purple-sound-569"
primary_region = "yyz"

[http_service]
  internal_port = 8080
  force_https = true
  auto_stop_machines = true
  auto_start_machines = true
```
This step should be clear on what needs to be changed in the file, or the example should look like their file?

Author: i must've been on an old CLI  version, so i've updated my tutorial.

> In order for internal requests to reach your app, an `internal_port` will need to be exposed in the app's fly.toml that was generated during launch. Here's what the _fly.toml_ might look like exposing port 8080:

Reviewer: You don't need a service defined at all for internal networking to work. It's only necessary when you're using Fly Proxy for routing, which is the _other_ (i.e. not-on-the-6PN-network) way to reach a Fly.io VM.

```toml
app = "my-app-name"
primary_region = "nrt"
kill_signal = "SIGINT"
kill_timeout = "5s"

[[services]]
  protocol = "tcp"
  internal_port = 8080 # <-- this is the important line
  processes = ["app"]

# remaining contents omitted
```

#### Install dig and/or curl in your Dockerfile

To test the internal network we need tool for that. By default dig and curl won't be installed on a machine. Just add `dnsutils` (for dig) and `curl` to your Dockerfile so you can access them later. Here's the line to add to your the Dockerfile:

```docker
RUN apt-get update -y && \
  apt-get install -y curl dnsutils \
  && apt-get clean && rm -f /var/lib/apt/lists/*_*
```

#### Deploy the app

> If you haven't already, run `fly deploy` and make sure the app is running. If you're using the [hello-fly](https://github.com/fly-apps/hello-fly) example then `service.port`'s were set in your fly.toml. You can  visit https://[[ your app name]].fly.dev and see a "Welcome to Fly" message.

Reviewer 2: ```suggestion
If you haven't already, run `fly deploy` and make sure the app is running. If you're using the [hello-fly](https://github.com/fly-apps/hello-fly) example, then `service.port`'s were set in your fly.toml. You can visit https://[[ your app name]].fly.dev and see a "Welcome to Fly.io" message.
```

> If you haven't already, run `fly deploy` and make sure the app is running. If you're using the [hello-fly](https://github.com/fly-apps/hello-fly) example then `service.port`'s were set in your fly.toml. You can  visit https://[[ your app name]].fly.dev and see a "Welcome to Fly" message.

Reviewer 2: `services-ports` are not set in the default `fly.toml` for apps v2 because it just uses `[http_service]`

#### Ping an internal address

Now it's time for the cool stuff.

1. SSH into the app `fly ssh console -a my-app-name`
2. Curl the app through the internal network:

```cmd
curl my-app-name.internal:8080
<!DOCTYPE html>
<html>
  <head>
    <title>Fly.io</title>
    <link rel='stylesheet' href='/stylesheets/style.css' />
  </head>
  <body>
    <h1>Fly.io (nrt)</h1>
    <p>Welcome to Fly.io (nrt)</p>
  </body>
</html>

# ↑↑↑ the same output as `curl https://my-app-name.fly.dev`

> ```

Reviewer 2: split this into command and output:

\```
curl my-app-name.internal:8080
\```

\```output
put output here
\```

> Nice! We've hit our app via the internal network.

Reviewer 2: ```suggestion
Nice! You've hit your app via the internal network.
```

> 

Reviewer 2: ```suggestion
To see a list of the apps on your private network:
```
suggest moving this line into text above the command

```cmd
# to see a list of apps on your private network:
dig -t txt _apps.internal +short
```

#### Tips

> - If your app is deployed to multiple regions, you can also hit a different instance of the app based on region using curl [[region]].[[your app name]].internal:8080

Reviewer: We use this convention: `<region>.<app-name>.internal:8080`

- In the example we pinged the same app we deployed, but try hitting another app in your network from within ssh console

> - Heads up: prefixing an internal network url request with http:// is fine, but obviously there aren't ssl certs issues for an private internal network so https:// won't work

Reviewer 2: ```suggestion
- Heads up: prefixing an internal network url request with `http://` is fine, but since ssl certs aren't issued for private internal networks, `https://` won't work
```
